src/core/caniuse.js

```diff
@@ -0,0 +1,289 @@
+document.head.appendChild(codeStyle);
+
+export const name = "core/caniuse";
+
+export function run(conf) {
+  normalizeConf(conf);
+  const { caniuse } = conf;
+  if (caniuse.feature) {
+    canIUse(caniuse.feature, document.querySelector(".head dl"), caniuse);
+  }
+  for (const el of document.querySelectorAll("section[data-caniuse]")) {
+    canIUse(el.dataset.caniuse, el.firstChild, caniuse);
+  }
+}
+
+/**
+ * MUTATES `conf.caniuse` object to hold normalized configurarion
+ * @param {Object|String} conf   configuration settings
+ */
+function normalizeConf(conf) {
+  const DEFAULTS = {
+    maxAge: 24 * 60 * 60 * 1000, // 24 hours (in ms)
+    browsers: ["chrome", "firefox", "safari", "edge"],
+    versions: 4,
+  };
+
+  if (!conf.caniuse) {
+    conf.caniuse = DEFAULTS;
+    return;
+  }
+  if (typeof conf.caniuse === "string") {
+    conf.caniuse = { feature: conf.caniuse, ...DEFAULTS };
+    return;
+  }
+  if (Array.isArray(conf.caniuse.browsers)) {
+    conf.caniuse.browsers = conf.caniuse.browsers.map(b => b.toLowerCase());
+  } else if (conf.caniuse.browsers !== "ALL") {
+    conf.caniuse.browsers = DEFAULTS.browsers;
+  }
+  if (conf.caniuse.maxAge === undefined) conf.caniuse.maxAge = DEFAULTS.maxAge;
+  if (!conf.caniuse.versions) conf.caniuse.versions = DEFAULTS.versions;
+}
+
+/**
+ * main canIUse function
+ * @param  {String} key              which api to look for
+ * @param  {Node} parent             adds table after parent
+ * @param {Object} conf              normalized respecConfig.caniuse
+ */
+async function canIUse(key, parent, conf) {
+  const url = `https://raw.githubusercontent.com/Fyrd/caniuse/master/features-json/${key}.json`;
+
+  const cache = await IDBCache("respec-caniuse");
+
+  // use data from cache data if valid and render
+  try {
+    const cachedStats = await cache.get(key);
+    if (cachedStats
+      && new Date() - new Date(cachedStats.cacheTime) < conf.maxAge) {
+      return showData(key, cachedStats.stats, conf, parent);
+    }
+  } catch (err) {
+    console.error("[core/caniuse]", err);
+  }
+
+  // otherwise fetch new data, cache and render
+  const placeholder = createPlaceholder(key, parent);
+  try {
+    const json = await getJson(url);
+    showData(key, json.stats, conf, parent);
+    cache.set(key, { stats: json.stats, cacheTime: new Date() })
+      .catch(err => console.error("[core/caniuse] (Could not cache)", err));
+  } catch (err) {
+    showError(err, key, placeholder);
+  }
+}
+
+/**
+ * creates a placeholder while the API fetches results
+ * @param  {String} key     API feature name
+ * @param  {Node} parent    where to create placeholder
+ * @return {Node}           inserted placeholder
+ */
+function createPlaceholder(key, parent) {
+  const placeholder = hyperHTML`
+    <div class="caniuse" id="${`caniuse-${key}`}">
+      <dt class="caniuse-title">
+        Can I Use this API? (${key})
+      </dt>
+      <dd>fetching data from caniuse.com... </dl>
```

Comment:
hyperHTML supports placeholders... I wonder if we can make better use of that functionality? I can't recall exactly what that looks like... but it will show the place holder until promises resolve.

Reply:
i'll look into it :)